pkg/reconciler/trigger/controller.go

```diff
@@ -111,8 +107,8 @@ func newController(ctx context.Context, cmw configmap.Watcher, drs *dataresidenc
 	// Watch brokers.
 	brokerinformer.Get(ctx).Informer().AddEventHandler(
 		cache.FilteringResourceEventHandler{
-			// Only care about brokers with the proper class annotation
-			FilterFunc: filterBroker,
+			// Only care about brokers with proper brokerclass
+			FilterFunc: reconcilerutils.BrokerClassFilter,
```

Comment:
Can you checked with Cong about this? Considering this case:

A broker started as a gcp broker, then the broker class is changed manually to MT broker, will the pub/sub topic for gcp broker leak?

Reply:
This event handler makes sure triggers's pubsub topic and subscription won't be leaked. Looking at the usage of filter function, as long as the updated broker is gcp broker either before the update or after the update, the trigger will be enqueued https://github.com/kubernetes/client-go/blob/b72204b2445de5ac815ae2bb993f6182d271fdb4/tools/cache/controller.go#L268. The filter function filters out triggers that's entirely unrelated to this broker.

Reply:
OK, it seems that both the original broker and trigger already have this filtering. The only additional filter here is in the brokercell. Makes sense to me.

Reply:
Changing the broker class is not allowed and will be rejected in the webhook, so the original question does not make sense any more.